core/src/main/scala/chisel3/Module.scala

```diff
@@ -251,7 +251,7 @@ package experimental {
 
     /** Legalized name of this module. */
     final lazy val name = try {
-      Builder.globalNamespace.name(desiredName)
+      if(this.isInstanceOf[ModuleAspect]) desiredName else Builder.globalNamespace.name(desiredName)
```

Comment:
A comment explaining why this is bypassing the normal naming process might be useful here?